src/renderers/dom/client/eventPlugins/BeforeInputEventPlugin.js

```diff
@@ -457,20 +457,21 @@ var BeforeInputEventPlugin = {
     nativeEvent,
     nativeEventTarget
   ) {
-    return [
-      extractCompositionEvent(
-        topLevelType,
-        targetInst,
-        nativeEvent,
-        nativeEventTarget
-      ),
-      extractBeforeInputEvent(
-        topLevelType,
-        targetInst,
-        nativeEvent,
-        nativeEventTarget
-      ),
-    ];
+    var compose = extractCompositionEvent(
+      topLevelType,
+      targetInst,
+      nativeEvent,
+      nativeEventTarget
+    );
+
+    var before = extractBeforeInputEvent(
+      topLevelType,
+      targetInst,
+      nativeEvent,
+      nativeEventTarget
+    );
+
+    return compose && before ? [compose, before] : compose ? compose : before;
```

Comment:
I'm surprised that changing the return type doesn't break any tests, do you know if there are cases in our test suite that returns `[null, null]`?